layers/+spacemacs/spacemacs-defaults/funcs.el

```diff
@@ -660,7 +660,9 @@ buffer in a split window."
       ('frame (select-frame (make-frame))))
     ;; Prompt to save on `save-some-buffers' with positive PRED
     (with-current-buffer newbuf
-      (setq-local buffer-offer-save t))
+      (progn
+        (setq-local buffer-offer-save t)
+        (funcall (default-value 'major-mode))))
```

Comment:
Instead of having to set the default value of the `major-mode` in `customize`.

It's probably better to define a separate variable for the new-empty-buffer function.

Above the function definition: 
https://github.com/syl20bnr/spacemacs/blob/46100767091a73f5744cc45fc2cacb56778329a9/layers/%2Bspacemacs/spacemacs-defaults/funcs.el#L648

Add a variable definition:
```elisp
(defvar spacemacs-new-empty-buffer-major-mode nil
  "Set the major mode for a new empty buffer.")
```

Then replace:
```elisp
        (funcall (default-value 'major-mode))))
```
with:
```elisp
      (when spacemacs-new-empty-buffer-major-mode
        (funcall spacemacs-new-empty-buffer-major-mode)))
```

And update the `spacemacs/new-empty-buffer` functions docstring to:
```text
  "Create a new buffer called untitled(<n>).
A SPLIT argument with the value: `left', `below', `above' or `right',
opens the new buffer in a split window.
If the variable `spacemacs-new-empty-buffer-major-mode' has been set,
then apply that major mode to the new buffer."
```

Now the major mode can be set from the `dotspacemacs/user-config` section of `.spacemacs`, with:
```elisp
    (setq spacemacs-new-empty-buffer-major-mode 'text-mode)
```

Reply:
The reason I thought to use `major-mode` is that the it would use the same mechanism as standard Emacs (without Spacemacs), but if a variable for a user-chosen function is preferred, I'll change it.

I'll update the PR.